libs/framework/src/bundle_context.c

```diff
@@ -839,7 +841,28 @@ static void celix_bundleContext_removeServiceTrackerTracker(void *data) {
     free(tracker);
 }
 
-static void celix_bundleContext_stopTrackerInternal(bundle_context_t *ctx, long trackerId, bool async, void *doneData, void (*doneCallback)(void* doneData)) {
+static void celix_bundleContext_waitForUnusedServiceTracker(celix_bundle_context_t* ctx,
+                                                            celix_bundle_context_service_tracker_entry_t* trkEntry) {
+    // busy wait till the tracker is not used anymore
+    // note that the use count cannot be increased anymore, because the tracker is removed from the map
+    struct timespec start = celix_gettime(CLOCK_MONOTONIC);
+    int logCount = 0;
+    while (__atomic_load_n(&trkEntry->useCount, __ATOMIC_RELAXED) > 0) {
```

Comment:
Thanks. I assumed ordering was not relevant, because a count increased only happened in the shared lock. 
But indeed this check and thus the destroy of the tracking needs to be ordered after the use count decrease.